src/world/Inventory/ItemContainer.cpp

```diff
@@ -67,12 +67,12 @@ const Sapphire::ItemMap& Sapphire::ItemContainer::getItemMap() const
 
 int8_t Sapphire::ItemContainer::getFreeSlot()
 {
-  for( uint16_t slotId = 0; slotId < m_size; slotId++ )
+  for( uint8_t slotId = 0; slotId < m_size; slotId++ )
```

Comment:
perhaps best to declare this as int8_t and remove the static_cast to int8_t below